### PR TITLE
twistday reminder with most of functionality

### PR DESCRIPTION
**Pull request by slr**
Twistday is day when you sent your first post. it is like a birthday in Twisterverse. looks like many of us want to celebrate it annually and congratulate others.

we have here almost finished Twistday Reminder module for `home.html` with related options in `options.html`. it observe peers which you are following and shows a list of today's luckies and yet another one for upcoming dates.

things TODO:
- [x] fix and localize strings
- [x] fix it's creepy appearance in `nin` theme. it has some differencies in realization of responsive design so it is not so easy to show it rightly ~~(and it's too late currently in my local area so excuse me and wait a bit)~~
- [x] add modal that would show post after a click on list item
- [ ] add notifications

~~it is not pull request yet, just for demo purposes and for following conversation only.~~

### CODE REVIEW

**Comment by slr** [comment]
I doubt about usefulness of notifications for this and it is may be added later anyway. so it's current status is _done_.

Who's celebrating Twisterday
Who's celebrating Twisterday
Twisterday Reminder options

**Comment by miguelfreitas** [comment]
@slr hmm... i'm not meant to discourage you, but not yet convinced this is such an important feature to merge... 
just to give a small example: what about real birthdays? of course a new field would need to be added, but if someone asked me what i'd think people would prefer to be reminded i'd guess that socially it would be more widely used.

**Comment by miguelfreitas** [comment]
@slr another idea, if you feel you might like to code new twister-html stuff: i've been cooking on my head, not entirely conceived though, a "group chat" idea. we may partner on development of that, for example, i do the core and you do the html ;-)

**Comment by slr** [comment]
hey, do not doubt please!

first, it already received strong positive reception from existing audience — e.g. see @myleneb's [reaction](https://twisterio.com/post/myleneb/2025):

> `@tasty but .. but.. ! that's so great thanks !!! <3 @rysiek`

or `@rysiek`'s [one](https://twisterio.com/post/rysiek/952):

> `@tasty that is SOOO COOOL! #Twistday`

by the way I was inspired by his [request](https://twisterio.com/post/rysiek/934) of bot for this:

> `Oh, I missed my #TwisterBirthday, it was Feb 8! Need to remember that. Can we have a bot that would inform about TwisterBirthdays? :)`

second, when you build things like Twister you build a new society actually. and in our case it is partly of a secret nature. in several kinds of organizations of this type when a newcomer arrives he recieves new name and thereby he is reborn at that moment. it is important thing to get dedication. and even if we are not a sect we can have some traditions.

and, finally, it is an optional thing. you provide benefits when you provide options to people. I believe we should feel free to add that field with actual date too and option to include observing of it to the reminder or not.

so, in short, it is all about popularization of Twister.

about goup chat idea I got to say only 'sure! why not?'

**Comment by dryabov** [comment]
@miguelfreitas I would add birthday field to profile and join date (when user was registered in the blockchain) to Twister RPC API (seems currently there is no easy way to get it using existing API).

**Comment by RealVegOs** [comment]
Of course it is not essential for Twister, it's more a fun factor. But as an option I doesn't annoy anybody and it is something other platforms don't have. 

Same refers to a chat function, at least for the networks I know. I wonder if this would change people's twist behavior and result in less twists. On the other hand it could attract new users.

Next would be video chat?

Please keep as much as possible as an option. I'm an old man and the world develops too fast! I'm a text guy.

**Comment by slr** [comment]
@dryabov we have tons of accounts which registered by squatters and currently being in inactive stage. obviously, date when they was registered in the blockchain is cannot be used for Twistday (see it's definition above).

**Comment by shaman** [comment]
In my opinion the minimalism is most important feature. This is especially true for microblog services. But custom options is not bad. I think it's time to create something like extensions API.

**Comment by miguelfreitas** [comment]
ok, ok, if people like the twistday lets give it a try ;-)

@shaman i agree with the minimalist + must be pleasant to use. imho a lot of people like wandering hours with smartphones without realizing what they love is the "moving fingers and seeing the phone reacting". no matter what are they effectively doing (reading, scrolling, playing, seeing pictures...) there is a subconscious rewarding due to the eye candy.

@shaman the problem with extensions API is security. one may code an extension to read your private key. now, we may all publicly review this patch before merging, so we make sure it is no evil...

**Comment by dryabov** [comment]
@slr Does Twistday limited to accounts a user follow? If yes, any date (registration one or first twist) may be used. If no, there will be a long queue of users per day in Twistday feed.

Anyway, my comment was not directly related to this patch, I'd like to see suggested fields (registration date and birthday) in user's profile. And it would be nice to see Birthday feed (based on users you follow) in addition to Twistday feed, just to send a cake picture on birthday. And it'll be possible to make a bot to send "Happy Birthday" messages to all twister users.

**Comment by miguelfreitas** [comment]
@dryabov the patch uses "max_id:0" to obtain the timestamp of the first twist.

**Comment by slr** [comment]
@dryabov yes, it is limited this way. no, any date is not may be used.

**Comment by shaman** [comment]
Looks good, but how about hiding the "Who's celebrating Twistday" text if celebrators are not found?
